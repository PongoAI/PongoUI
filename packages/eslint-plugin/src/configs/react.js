/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
