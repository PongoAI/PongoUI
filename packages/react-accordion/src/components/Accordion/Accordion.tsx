import { Accordion as FluentUIAccordion } from '@fluentui/react-accordion';

/**
 * The Accordion component is used to show and hide nested information.
 */
export const Accordion = FluentUIAccordion;
