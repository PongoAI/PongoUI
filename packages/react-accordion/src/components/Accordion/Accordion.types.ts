import type {
  AccordionProps as FluentAccordionProps,
  AccordionState as FluentAccordionState,
  AccordionSlots as FluentAccordionSlots,
} from '@fluentui/react-accordion';

export type AccordionSlots = FluentAccordionSlots;

export type AccordionProps = FluentAccordionProps;

export type AccordionState = FluentAccordionState;
