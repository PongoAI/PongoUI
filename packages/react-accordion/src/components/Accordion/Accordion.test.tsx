import * as React from 'react';
import { Accordion } from './Accordion';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('Accordion', () => {
  isConformant({
    Component: Accordion,
    displayName: 'Accordion',
    disabledTests: ['consistent-callback-args', 'make-styles-overrides-win'],
  });

  describe('Snapshot tests', () => {
    it('renders a basic Accordion', () => {
      const result = render(<Accordion />);

      expect(result.baseElement).toMatchSnapshot();
    });
  });
});
