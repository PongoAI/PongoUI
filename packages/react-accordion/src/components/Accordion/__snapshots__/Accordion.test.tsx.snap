// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Snapshot tests renders a basic Accordion 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;
